content/en/service_management/app_builder/build.md

```diff
@@ -1,5 +1,294 @@
+
+### Conditional queries
+
+You can set a condition that must be met before a query can run. To set a query, enter an expression in the **Condition** field in the **Advanced** section of the query. The condition must evaluate to true before the query runs. For example, if you want a given query to run only if a UI component named `select0` exists and is not empty, you can use the expression `${select0.value && select0.value.length > 0}`.
+
+### Post query transformation
+
+You can perform a post query transformation to simplify or transform the output of a query. Add a post query transformation in the **Advanced** section of a query.
+
+For example, the Slack _List Channels_ action returns an array of dictionaries containing the ID and name for each channel. To discard the IDs and return only an array of names, add the following query transformation:
+
+{{< code-block lang="js" collapsible="false" >}}
+// Use `outputs` to reference the query's unformatted output.
+// TODO: Apply transformations to the raw query output
+arr = []
+object = outputs.channels
+for (var item in object) {
+    arr.push(object[item].name);
+}
+
+return arr
+}
+{{< /code-block >}}
+
+### Error notifications
+
+To display a toast to the user when the system returns an error, toggle **Show Toast on Errors** in the **Advanced** section of a query.
+
+### Confirmation prompts
+
+To prompt a user for confirmation before the query runs, toggle the **Requires Confirmation** option in the **Advanced** section of a query.
+
+## Variables
+
+Use app variables to pass data from one part of your app to another. Additionally, you can use app variables to pass in data from your dashboard using dashboard template variables.
+
+Variables are enclosed in braces, proceeded by a dollar sign (`${}`). To use a variable, use the query or UI component name and access the child fields using dot notation. For example, if you have a select component named `select0` and you want to access its default value field in a query, use the syntax `${select0.defaultValue}`. If you're not sure what to enter as a variable, type `${` to open a suggestions menu with all available variables.
+
+{{< img src="service_management/app_builder/select-variable.mp4" alt="If you're not sure what to enter as a variable, type ${ to open a suggestions menu with all available variables" video=true >}}
+
+For more information using dashboard template variables, see [Embedding apps in dashboards][3].
+
+## Preview an app
+
+Click the **Preview** button to preview your app. Preview mode allows you to view the app from the user's perspective. Use preview mode to interact with the app UI and test your queries. When you're done, click **Edit** to return to the app builder.
+
+## Further reading
+
+{{< partial name="whats-next/whats-next.html" >}}
+
+[1]: https://app.datadoghq.com/app-builder/apps/
+[2]: https://app.datadoghq.com/app-builder/blueprints
+[3]: /service_management/app_builder/embedded_apps
```

Comment:
I want to link off to Bryce's doc here, but I don't know the URL yet, so this is a placeholder